packages/realm_dart/test/dynamic_realm_test.dart

```diff
@@ -222,6 +226,95 @@ void main() {
     expect(actualDynamic.nullableIntProp, expected.nullableIntProp);
     expect(actualDynamic.decimalProp, expected.decimalProp);
     expect(actualDynamic.nullableDecimalProp, expected.nullableDecimalProp);
+    expect(actualDynamic.realmValueProp, expected.realmValueProp);
+  }
+
+  void _validateDynamicSetters(RealmObject actual, AllTypes expected) {
+    actual.realm.write(() {
+      actual.dynamic.set('stringProp', 'updated abc');
+      actual.dynamic.set('nullableStringProp', 'updated abc');
+
+      actual.dynamic.set('boolProp', false);
+      actual.dynamic.set('nullableBoolProp', false);
+
+      actual.dynamic.set('dateProp', DateTime(1999));
+      actual.dynamic.set('nullableDateProp', DateTime(1999));
+
+      actual.dynamic.set('doubleProp', -999.111);
+      actual.dynamic.set('nullableDoubleProp', 111.999);
+
+      actual.dynamic.set('objectIdProp', ObjectId());
+      actual.dynamic.set('nullableObjectIdProp', ObjectId());
+
+      actual.dynamic.set('uuidProp', Uuid.v4());
+      actual.dynamic.set('nullableUuidProp', Uuid.v4());
+
+      actual.dynamic.set('intProp', 42);
+      actual.dynamic.set('nullableIntProp', -42);
+
+      actual.dynamic.set('decimalProp', Decimal128.fromDouble(500));
+      actual.dynamic.set('nullableDecimalProp', Decimal128.infinity);
+
+      actual.dynamic.set('realmValueProp', RealmValue.from([true, 5]));
+    });
+
+    expect(actual.dynamic.get('stringProp'), isNot(expected.stringProp));
```

Comment:
Would it not be preferable to test against the exact updated value instead of `isNot` here?

Reply:
Sure, we could also do that - I was mostly lazy here.

Reply:
Then we can catch potential incorrect updates 👍